component/loki/source/file/decompresser.go

```diff
@@ -0,0 +1,311 @@
+package file
```

Comment:
This file is from Promtail, right? I think it's ok to merge as-is but I'd be interested in doing some refactoring later to clean it up a bit.

Reply:
Yes, I refrained from making any changes for now. As you suggested I will add a comment to all copied files and start seeing how we can simplify things we import.